In [81]:
## 1.importing numpy(to create array and get basic information like mean, std of data) and stats package( for statistical operations)
import numpy as np
import scipy.stats as stats

In [83]:
## 2. creating array using numpy
data=np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])
data

array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.2 ,
       1.33, 1.18, 1.22, 1.29])

In [189]:
## 3.calculating sample mean 
x_bar=np.mean(data)
x_bar

1.2386666666666666

In [87]:
## 3.calculating sample standard deviation
s=np.std(data,ddof=1)
s

0.19316412956959936

In [89]:
## 3.calculating size of the data 
n=np.size(data)
n

15

####  the sample size is less than 30..therefore we use t-test

## a. Build 99% Confidence Interval Using Sample Standard Deviation

In [185]:
## 4.finding t critical value
## 99% confidence..alpha=1-CI
alpha=0.01
t_critic=stats.t.ppf(1-alpha/2,n-1)
t_critic

2.976842734370834

In [152]:
## 5.calculating error bound using  formula : t_critical*(sample std/sqrt(size_of_data))
error_bound=t_critic*(s/np.sqrt(n))
error_bound

0.14846932822817596

In [154]:
## 6. calculating lower bound and upper bound by subtract and adding sample_mean to error_bound.
lower_bound_of_pop_mean=x_bar-error_bound
lower_bound_of_pop_mean

1.0901973384384906

In [156]:
higher_bound_of_pop_mean=x_bar+error_bound
higher_bound_of_pop_mean

1.3871359948948425

#### pop_mean rangers from 1.09 to 1.39
#### From this we can conclude that: 1.09 to 1.39 millian number of characters printed before the print-head fails 

## b. Build 99% Confidence Interval Using Known Population Standard Deviation

In [197]:
## when pop_std is given we use z_test instead of t-test
pop_std=0.2

In [199]:
z_critic=stats.norm.ppf(1-alpha/2)
z_critic

2.5758293035489004

In [201]:
error_bound=z_critic*(pop_std/np.sqrt(n))
error_bound

0.13301525327090588

In [203]:
## 6. calculating lower bound and upper bound by subtract and adding sample_mean to error_bound.
lower_bound_of_pop_mean=x_bar-error_bound
lower_bound_of_pop_mean

1.1056514133957607

In [205]:
higher_bound_of_pop_mean=x_bar+error_bound
higher_bound_of_pop_mean

1.3716819199375725

#### pop_mean rangers from 1.11 to 1.37
#### From this we can conclude that: 1.11 to 1.37 millian number of characters printed before the print-head fails¶